frontend/types/types.ts

```diff
@@ -1,77 +1,143 @@
+  | Schema.Attribute.GetDynamicZoneValue<TAttribute>
+  | Schema.Attribute.GetEnumerationValue<TAttribute>
+  | Schema.Attribute.GetEmailValue<TAttribute>
+  | Schema.Attribute.GetFloatValue<TAttribute>
+  | Schema.Attribute.GetIntegerValue<TAttribute>
+  | Schema.Attribute.GetJsonValue<TAttribute>
+  | Schema.Attribute.GetMediaValue<TAttribute>
+  | Schema.Attribute.GetPasswordValue<TAttribute>
+  | Schema.Attribute.GetRelationValue<TAttribute>
+  | Schema.Attribute.GetRichTextValue<TAttribute>
+  | Schema.Attribute.GetStringValue<TAttribute>
+  | Schema.Attribute.GetTextValue<TAttribute>
+  | Schema.Attribute.GetUIDValue<TAttribute>
+  | Schema.Attribute.GetDateValue<TAttribute>
+  | Schema.Attribute.GetDateTimeValue<TAttribute>
+  | Schema.Attribute.GetTimeValue<TAttribute>
+  | Schema.Attribute.GetTimestampValue<TAttribute>,
+  TGuard
+>;
+
+type OriginalGetAll<TSchemaUID extends UID.Schema> = Utils.Get<
+  Schema.Schemas[TSchemaUID],
+  "attributes"
+>;
+type OriginalGet<
+  TSchemaUID extends UID.Schema,
+  TKey extends OriginalGetKeys<TSchemaUID>,
+> = Utils.Get<OriginalGetAll<TSchemaUID>, TKey>;
+
+type OriginalGetRequiredKeys<TSchemaUID extends UID.Schema> =
+  Utils.Object.KeysBy<
+    OriginalGetAll<TSchemaUID>,
+    Schema.Attribute.Required,
+    string
+  >;
+type OriginalGetOptionalKeys<TSchemaUID extends UID.Schema> =
+  Utils.Object.KeysExcept<
+    OriginalGetAll<TSchemaUID>,
+    Schema.Attribute.Required,
+    string
+  >;
+
+type OriginalGetKeys<TSchemaUID extends UID.Schema> =
+  keyof OriginalGetAll<TSchemaUID> & string;
+
+export type GetValues<TSchemaUID extends UID.Schema> = {
+  [TKey in OriginalGetOptionalKeys<TSchemaUID>]?: GetValue<
+    OriginalGet<TSchemaUID, TKey>
+  >;
+} & {
+  [TKey in OriginalGetRequiredKeys<TSchemaUID>]-?: GetValue<
+    OriginalGet<TSchemaUID, TKey>
+  >;
+};
+
+/**
  * Attribute.GetValue override with extended values
  *
  * Fallback to unknown if never is found
  */
 
-export type GetValue<TAttribute extends Attribute.Attribute> = Utils.Expression.If<
-Utils.Expression.IsNotNever<TAttribute>,
-Utils.Expression.MatchFirst<
-  [
-    // Relation
-    [
-      Utils.Expression.Extends<TAttribute, Attribute.OfType<'relation'>>,
-      any // Response<RelationTarget>
-    ],
-    // DynamicZone
-    [
-      Utils.Expression.Extends<TAttribute, Attribute.OfType<'dynamiczone'>>,
-      TAttribute extends Attribute.DynamicZone<infer TComponentsUIDs>
-        ? Array<
-            // Extract tuple values to a component uid union type
-            Utils.Array.Values<TComponentsUIDs> extends infer TComponentUID
-              ? TComponentUID extends Common.UID.Component
-                ? GetValues<TComponentUID> & { __component: TComponentUID }
+export type GetValue<TAttribute> = TAttribute extends Schema.Attribute.Attribute
+  ? Utils.If<
+      Utils.IsNotNever<TAttribute>,
+      Utils.MatchFirst<
+        [
+          [Utils.Extends<TAttribute, Schema.Attribute.OfType<"relation">>, any],
```

Comment:
Hi, here is a solution to be more specific for relations.

```suggestion
          [
            Utils.Extends<TAttribute, Schema.Attribute.OfType<"relation">>,
            TAttribute extends Schema.Attribute.RelationWithTarget<infer _TRelationKind, infer TTargetUID>
              ? TTargetUID extends UID.ContentType
                ? TAttribute["target"] extends TTargetUID
                  ? Array<GetValues<TTargetUID>>
                  : never
                : never
              : any
          ],
```

Reply:
Thanks!